lib/turn.rb

```diff
@@ -28,6 +25,11 @@ def validate_move(move, valid_columns)
     move
   end
 
+  def column_is_full?(move, columns)
+    column_index = columns.index(move)
+    @board.board_grid[0][column_index].state != "."
```

Comment:
This is great! It's obvious now that I read it but checking for first row aka board_grid[0] to not have a "." is a simple and elegant way to check for fullness of the row. nice work!